modules/Emails/EmailsController.php

```diff
@@ -813,12 +814,14 @@ protected function userIsAllowedToSendEmail($requestedUser, $requestedInboundEma
 
         $inboundEmailStoredOptions = $requestedInboundEmail->getStoredOptions();
 
-        // if group email account, check that user is allowed to use group email account
-        if ($requestedInboundEmail->isGroupEmailAccount()) {
-            if ($inboundEmailStoredOptions['allow_outbound_group_usage'] === true) {
-                $hasAccessToInboundEmailAccount = true;
-            } else {
-                $hasAccessToInboundEmailAccount = false;
+        if (!$hasAccessToInboundEmailAccount) {
```

Comment:
Logically this doesn't make sense. retrieveAllByGroupIdWithGroupAccounts is supposed to retrieve all group (and personal) accounts the user has access but right now it's broken and retrieves all group accounts regardless if the user has access or not.

So, with the current broken behavior, $hasAccessToInboundEmailAccount is always true for group inbound accounts, so the next if is equivalient to `if (false)`.

If retrieveAllByGroupIdWithGroupAccounts performed as documented, `if (!$hasAccessToInboundEmailAccount)` means the user really doesn't have access to the inbound group account, but we allow access to the outbound account if the inbound account allows outbound access to their group members (although the user doesn't belong to this group).

Reply:
I applied this to 7.9.7 and still doesn't work.
Configured both system and personal outgoing email, tested ok. But got error when trying to compose, then send.
UI showed: "Error Sending Email. Please contact your administrator for assistance."
Log showed: [SECURITY] User is not allowed to send email
Reason was:
User is not allowed to use outbound email.
[SECURITY] User XXX attempted to send an email using incorrect email account settings in which they do not have access to.
Please help help !